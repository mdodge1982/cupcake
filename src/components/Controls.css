.Controls {
	width: 100%;
	text-align: center;
	padding: 10px;
}
.shoot {
	border-width: 5px;
	border-radius: 20px;
	height: 35px;
	width: 35px;
}

@media screen and ( max-height: 500px ){
	.Controls {
		display: none;
	}
}
