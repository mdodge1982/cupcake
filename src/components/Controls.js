import React, {Component} from 'react';
import ArrowContainer from '../containers/ArrowContainer';
import Instructions from '../components/Instructions';
import './Controls.css';

class Controls extends Component {
	render() {
		return (
			<div className="Controls">
				<ArrowContainer direction="left"/>
				<button className="shoot" onClick={() => this.props.addFrosting()}></button>
				<ArrowContainer direction="right"/>
				<Instructions small={true} />
			</div>
		);
	}
}

export default Controls;
