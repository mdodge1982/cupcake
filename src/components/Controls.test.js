import React from 'react';
import ReactTestUtils from 'react-dom/test-utils';
import {Provider} from 'react-redux';
import {createStore} from 'redux';
import renderer from 'react-test-renderer';
import Controls from './Controls';

const reducer = (state = {}) => state;
const store = createStore(reducer);

test('Snapshot test', () => {
	const component = renderer.create(
		<Provider store={store}>
			<Controls />
		</Provider>
	);
	let tree = component.toJSON();
	expect(tree).toMatchSnapshot();

	tree.props.wait = 18;
	tree = component.toJSON();
	expect(tree).toMatchSnapshot();
});

test('onClick test', () => {
	const mockFn = jest.fn();
	const component = ReactTestUtils.renderIntoDocument(
		<Provider store={store}>
			<Controls addFrosting={mockFn} />
		</Provider>
	);
	const button = ReactTestUtils.findRenderedDOMComponentWithClass(component,'shoot');
	ReactTestUtils.Simulate.click(button);
	expect(mockFn.mock.calls.length).toBe(1);
});
