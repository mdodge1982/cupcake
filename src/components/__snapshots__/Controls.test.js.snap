// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot test 1`] = `
<div
  className="Controls"
>
  <button
    className="Arrow left"
    onMouseDown={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchStart={[Function]}
  />
  <button
    className="shoot"
    onClick={[Function]}
  />
  <button
    className="Arrow right"
    onMouseDown={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchStart={[Function]}
  />
  <p
    className="Instructions small"
  >
    <strong>
      HINT:
    </strong>
     Use the 
    <strong>
      left
    </strong>
     and 
    <strong>
      right
    </strong>
     arrow keys to move your pastry bag, and the 
    <strong>
      spacebar
    </strong>
     (or click anywhere on the board) to shoot frosting.
  </p>
</div>
`;
